service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/disk/ControlledAzureDiskHandler.java

```diff
@@ -31,4 +32,8 @@ public WsmResource makeResourceFromDb(DbResource dbResource) {
             .build();
     return resource;
   }
+
+  public String generateCloudName(UUID workspaceUuid, String resourceName) {
```

Comment:
Once your PR is in, you can advertise this new api in the slack group to see if the azure folks want to implement this for their resources.